packages/customize-widgets/src/components/inserter/use-inserter.js

```diff
@@ -1,157 +1,33 @@
-	}
-
-	const [ isInserterOpened, setIsInserterOpened ] = useState( () =>
-		inserterRef.current.expanded()
-	);
-
-	// This is useful when the expanded state is controlled from outside React,
-	// like when handling multiple outer sections above.
-	useEffect( () => {
-		const inserter = inserterRef.current;
-
-		function syncExpandedStateToComponentState( expanded ) {
-			setIsInserterOpened( expanded );
-		}
-
-		inserter.expanded.bind( syncExpandedStateToComponentState );
-
-		return () => {
-			inserter.expanded.unbind( syncExpandedStateToComponentState );
-		};
-	}, [] );
-
-	useEffect(
-		function syncOpenedStateToOuterSectionExpandedState() {
-			const inserter = inserterRef.current;
-
-			if ( isInserterOpened ) {
-				const contentContainer = inserter.contentContainer[ 0 ];
-				const activeElementBeforeExpanded =
-					contentContainer.ownerDocument.activeElement;
-
-				inserter.expand( {
-					completeCallback() {
-						// We have to do this in a "completeCallback" or else the elements will not yet be visible/tabbable.
-						// The first one should be the close button,
-						// we want to skip it and choose the second one instead, which is the search box.
-						const searchBox = focus.tabbable.find(
-							contentContainer
-						)[ 1 ];
-						if ( searchBox ) {
-							searchBox.focus();
-						}
-					},
-				} );
-
-				return () => {
-					if ( inserter.expanded() ) {
-						const activeElement =
-							contentContainer.ownerDocument.activeElement;
-
-						// Return back the focus when closing the inserter.
-						// Only do this if the active element which triggers the action is inside the inserter,
-						// (the close button for instance). In that case the focus will be lost.
-						// Otherwise, we don't hijack the focus when the user is focusing on other elements
-						// (like the quick inserter).
-						if ( contentContainer.contains( activeElement ) ) {
-							inserter.collapse( {
-								completeCallback() {
-									// Return back the focus when closing the inserter.
-									if ( activeElementBeforeExpanded ) {
-										activeElementBeforeExpanded.focus();
-									}
-								},
-							} );
-						}
-					}
-				};
-			}
-
-			if ( inserter.expanded() ) {
-				inserter.collapse();
-			}
-		},
-		[ isInserterOpened ]
+export default function useInserter( inserter ) {
+	const [ isInserterOpened, setIsInserterOpened ] = useState(
+		() => inserter.isOpen
 	);
 
-	// Handle closing the inserter when pressing the Escape key.
 	useEffect( () => {
-		if ( isInserterOpened ) {
-			const ownerWindow =
-				inserterRef.current.contentContainer[ 0 ].ownerDocument
-					.defaultView;
-
-			function handlePressEscapeKey( event ) {
-				if ( event.keyCode === ESCAPE || event.code === 'Escape' ) {
-					event.stopPropagation();
-
-					setIsInserterOpened( false );
+		return inserter.subscribe( setIsInserterOpened );
+	}, [ inserter ] );
+
+	return [
+		isInserterOpened,
+		useCallback(
```

Comment:
Returning a `setIsInserterOpened` function here doesn't seem to give us much compared to just passing `openInserter` and `closeInserter` down to `Header`. But up to you—I don't mind.

Reply:
It just feels more natural to return a React-style state tuple 😝. The sole purpose of this hook is to bridge between the BackBone state and the React state, BackBone -> React.

Hence, IMO, the input of the hook should be more like a BackBone thing (InserterOuterSection), and the output of the hook should be more like a React thing (a state tuple just like `useState`).

Reply:
You're the boss!

Reply:
Probably I'll change my mind a month later though 😆